StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheet+API.swift

```diff
@@ -582,44 +591,103 @@ extension PaymentSheet {
+                    // Server-side confirmation
+                    func handleStatus(status: STPPaymentHandlerActionStatus, error: Error?) {
+                        switch status {
+                        case .succeeded:
+                            deferredIntentContext.completion(.completed)
+                        case .canceled:
+                            deferredIntentContext.completion(.canceled)
+                        case .failed:
+                            let error = error ?? PaymentSheetError.unknown(debugDescription: "Unknown error occured while handling intent next action")
+                            deferredIntentContext.completion(.failed(error: error))
+                        @unknown default:
+                            deferredIntentContext.completion(.failed(error: PaymentSheetError.unknown(debugDescription: "Unrecognized intent status")))
+                        }
+                    }
 
-            guard let paymentMethod = paymentMethod else {
-                let paymentMethodCreationError = PaymentSheetError.unknown(
-                    debugDescription: "Failed to create a new payment method."
-                )
-                completion(.failed(error: paymentMethodCreationError))
-                return
-            }
+                    switch deferredIntentContext.intentConfig.mode {
+                    case .payment:
+                        deferredIntentContext.paymentHandler.handleNextAction(forPayment: clientSecret,
+                                                                        with: deferredIntentContext.authenticationContext,
+                                                                        returnURL: deferredIntentContext.configuration.returnURL,
+                                                                        completion: { status, _, error in
+                            handleStatus(status: status, error: error)
+                        })
+                    case .setup:
+                        deferredIntentContext.paymentHandler.handleNextAction(forSetupIntent: clientSecret,
+                                                                        with: deferredIntentContext.authenticationContext,
+                                                                        returnURL: deferredIntentContext.configuration.returnURL,
+                                                                        completion: { status, _, error in
+                            handleStatus(status: status, error: error)
+                        })
+                    }
+                } else {
+                    // Client-side confirmation
+                    let intent = try await deferredIntentContext.configuration.apiClient.retrieveIntent(for: deferredIntentContext.intentConfig,
+                                                                                                        withClientSecret: clientSecret)
```

Comment:
Let's add a comment indicating that this retrieval isn't strictly necessary and we could remove it in the future as an optimization